File: 10_Sorting/Insertion_Sort.java
package Sorting;

public class Insertion_Sort {
    public static void perform_insertion_sort(int [] arr)
    {
        int n=arr.length;
        for(int i=0;i<n;i++)
        {
            int temp=arr[i];
            int j=i;
            while(j>0 && arr[j-1]>temp) // We are traversing to place it in sorted array
            {
                arr[j]=arr[j-1];
                j--;
            }
            arr[j]=temp;
        }
    }
    public static void main(String args[])
    {
        int[] arr={5,9,3,1,2};
        perform_insertion_sort(arr);
        for(int i=0;i<arr.length;i++)
        {
            System.out.print(arr[i]+"\t");
        }
    }
}

File: 10_Sorting/insertion_sort.py
# select the first element from the unsorted array and insert it in sorted array at right position

def insertion_sort(arr):
    for i in range(1,len(arr)):
        j=i
        val=arr[i]
        while j>0 and arr[j-1]>val:
            arr[j]=arr[j-1]
            j=j-1
        arr[j]=val
    return arr


print(insertion_sort([4,9,2,1,2,32]))